src/OpenTelemetry/Internal/OpenTelemetrySdkEventSource.cs

```diff
@@ -350,13 +350,13 @@ public void LoggerProcessStateSkipped(string type, string reason)
 #if DEBUG
     public class OpenTelemetryEventListener : EventListener
     {
-        private readonly List<EventSource> eventSources = new();
+        private readonly Dictionary<string, EventSource> eventSources = new();
 
         public override void Dispose()
         {
-            foreach (EventSource eventSource in this.eventSources)
+            foreach (var kvp in this.eventSources)
```

Comment:
This shouldn't run in parallel with `OnEventSourceCreated` method as it could modify the dictionary while `Dispose` method is enumerating over it.

Reply:
> I'm not too concerned given it is just a debug utility.

Makes sense.